salt/matchers/compound_match.py

```diff
@@ -25,11 +25,15 @@ def match(tgt):
     '''
     nodegroups = __opts__.get('nodegroups', {})
     matchers = salt.loader.matchers(__opts__)
+    if 'minion_id' in __opts__:
+        minion_id = __opts__['minion_id']
+    else:
+        minion_id = __opts__['id']
```

Comment:
Is LBYL preferred or would

```
try:
    minion_id = __opts__['minion_id']
except KeyError:
    minion_id = __opts__['id']
```

be a better approach? Not that it would probably matter *here* but I'm pretty sure the EAFP version is slightly faster. As another alternative, I think `minion_id = __opts__.get('minion_id', __opts__['id'])` would also work, but not sure if thats funkier lookin' :man_shrugging:

Reply:
I like the __opts__.get approach better, cleaner.  Updated the PR.